### PR TITLE
[INTERNAL] Slacken the restriction on the qualified class from nucleotide_concept to alphabet_concept.

### LINKED ISSUE #359: qualified class should not be restricted to nucleotide alphabets
This especially hinders the gapped<alphabet> because it does not fulfil the nucleotide_concept as a `gap::GAP` does not have a complement. Hence, the quality_composition cannot be build over an gapped alphabet which would be desirable though IMPOV.

Also, the user might want to attach qualities to other strings..

### PR DESCRIPTION
**Pull request by smehringer**
Fixes #359

### CODE REVIEW

File: include/seqan3/alphabet/quality/qualified.hpp [hunk 1]
```diff
@@ -49,16 +49,20 @@
...
+ * \tparam sequence_alphabet_t Type of the alphabet; must satisfy seqan3::alphabet_concept.
+ * \tparam quality_alphabet_t  Type of the quality; must satisfy seqan3::quality_concept.
  * \implements seqan3::quality_concept
  *
- * This composition pairs a nucleotide alphabet with a quality alphabet. The rank values
- * correpsond to numeric values in the size of the composition, while the character values
- * are taken from the sequence alphabet and the phred values are taken from the quality
- * alphabet.
+ * This composition pairs an arbitrary alphabet with a quality alphabet. Note,
+ * that usually this is a nucleotide alphabet, e. g. a FASTQ sequence stored
```
Comment (h-2) [inline include/seqan3/alphabet/quality/qualified.hpp]: but you are not storing a sequence in a single object here, only two letter :nail_care:

File: include/seqan3/alphabet/quality/qualified.hpp [hunk 2]
```diff
@@ -49,16 +49,20 @@
...
+ * within a single object, but in certain cases it is desirable to also allow
+ * other alphabets. An example is a `gapped<dna4q>` alphabet that further allows
```
Comment (h-2) [inline include/seqan3/alphabet/quality/qualified.hpp]: This isn't about `gapped<dna4q>`, but about `qualified<gapped<dna4>>`.

**Comment by h-2** [comment]
> And either way, IMPOV the nucleotide concept is too restrictive?

Independent of the complement function, all nucleotides should be convertible to each other so `qualified<dna5>` can be a nucleotide, but `gapped<dna5>` can't.

File: include/seqan3/alphabet/quality/qualified.hpp [hunk 3]
```diff
@@ -49,20 +49,29 @@
...
+ * \tparam sequence_alphabet_t Type of the alphabet; must satisfy seqan3::alphabet_concept.
+ * \tparam quality_alphabet_t  Type of the quality; must satisfy seqan3::quality_concept.
  * \implements seqan3::quality_concept
  *
- * This composition pairs a nucleotide alphabet with a quality alphabet. The rank values
- * correpsond to numeric values in the size of the composition, while the character values
- * are taken from the sequence alphabet and the phred values are taken from the quality
- * alphabet.
+ * This composition pairs an arbitrary alphabet with a quality alphabet, where
+ * each alphabet character is stored together with it's quality score in a
```
Comment (eseiler) [inline include/seqan3/alphabet/quality/qualified.hpp]: its
it's is always short for it is

File: include/seqan3/alphabet/quality/qualified.hpp [hunk 4]
```diff
@@ -49,20 +49,29 @@
...
+ * single value. That way, you can can conveniently access the character and
+ *  score information at each position of the qualified-sequence.
+ * The use case that this was designed for is a nucleotide sequence with
+ * corresponding quality scores, e.g. obtained when reading in a FASTQ file
+ * of Illumina reads.
+ * The composition also allows to store quality scores for different or extended
+ * alphabets like a `qualified<char, phred42> or a `qualified<gapped<dna4>, phred42>`
```
Comment (eseiler) [inline include/seqan3/alphabet/quality/qualified.hpp]: `qualified<char, phred42>`
(closing `)

File: include/seqan3/alphabet/quality/qualified.hpp [hunk 5]
```diff
@@ -49,20 +49,29 @@
...
+ * The rank values correspond to numeric values in the size of the composition,
+ * while the character values are taken from the sequence alphabet and the phred
+ * values are taken from the quality alphabet.
  *
- * As with all `seqan3::cartesian_composition` s you may access the individual alphabet letters in
- * regular c++ tuple notation, i.e. `get<0>(t)` and objects can be brace-initialized
- * with the individual members.
+ * As with all `seqan3::cartesian_composition` s you may access the individual
+ * alphabet letters in regular c++ tuple notation, i.e. `get<0>(t)` and objects
+ * can be brace-initialized with the individual members.
```
Comment (h-2) [inline include/seqan3/alphabet/quality/qualified.hpp]: :gb: english please :nail_care:

**Pull request merged**